adapter/internal/oasparser/model/policy_container.go

```diff
@@ -0,0 +1,234 @@
+const (
+	policyValTypeString string = "String"
+	policyValTypeInt    string = "Integer"
+	policyValTypeBool   string = "Boolean" // TODO: check type names with APIM
+	policyValTypeArray  string = "Array"
+	policyValTypeMap    string = "Map"
+)
+
+// PolicyFlow holds list of Policies in a operation (in one flow: In, Out or Fault)
+type PolicyFlow string
+
+var (
+	policyInFlow    PolicyFlow = "request"
+	policyOutFlow   PolicyFlow = "response"
+	policyFaultFlow PolicyFlow = "fault"
+)
+
+// PolicyContainer holds the definition and specification of policy
+type PolicyContainer struct {
+	Specification PolicySpecification
+	Definition    PolicyDefinition
+}
+
+// PolicyContainerMap maps PolicyName -> PolicyContainer
+type PolicyContainerMap map[string]PolicyContainer
+
+// GetFormattedOperationalPolicies returns formatted, Choreo Connect policy from a user templated policy
+func (p PolicyContainerMap) GetFormattedOperationalPolicies(policies OperationPolicies) OperationPolicies {
+	fmtPolicies := OperationPolicies{}
+
+	inFlowStats := policies.In.getStats()
+	for i, policy := range policies.In {
+		if fmtPolicy, err := p.getFormattedPolicyFromTemplated(policy, policyInFlow, inFlowStats, i); err == nil {
+			fmtPolicies.In = append(fmtPolicies.In, fmtPolicy)
+		}
+	}
+
+	outFlowStats := policies.Out.getStats()
+	for i, policy := range policies.Out {
+		if fmtPolicy, err := p.getFormattedPolicyFromTemplated(policy, policyOutFlow, outFlowStats, i); err == nil {
+			fmtPolicies.Out = append(fmtPolicies.In, fmtPolicy)
+		}
+	}
+
+	faultFlowStats := policies.Out.getStats()
+	for i, policy := range policies.Fault {
+		if fmtPolicy, err := p.getFormattedPolicyFromTemplated(policy, policyFaultFlow, faultFlowStats, i); err == nil {
+			fmtPolicies.Fault = append(fmtPolicies.In, fmtPolicy)
+		}
+	}
+
+	return fmtPolicies
+}
+
+// getFormattedPolicyFromTemplated returns formatted, Choreo Connect policy from a user templated policy
+func (p PolicyContainerMap) getFormattedPolicyFromTemplated(policy Policy, flow PolicyFlow, stats map[string]policyStats, index int) (Policy, error) {
+	// using index i instead of struct to validate policy against multiple allowed and apply only first if multiple exists
+	spec := p[policy.PolicyName].Specification
+	if err := spec.validatePolicy(policy, flow, stats, index); err != nil {
+		loggers.LoggerAPI.Errorf("Operation policy validation failed, ignoring the policy \"%v\": %v", policy.PolicyName, err)
+		return policy, err
+	}
+
+	defRaw := p[policy.PolicyName].Definition.RawData
+	t, err := template.New("policy-def").Parse(string(defRaw))
+	if err != nil {
+		loggers.LoggerAPI.Errorf("Error parsing the operation policy definition \"%v\" into go template: %v", policy.PolicyName, err)
+		return Policy{}, err
+	}
+
+	var out bytes.Buffer
+	err = t.Execute(&out, policy.Parameters)
+	if err != nil {
+		loggers.LoggerAPI.Errorf("Error operation policy definition \"%v\": %v", policy.PolicyName, err)
```

Comment:
Use the formalized error log format..

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2656/commits/febace122f95f303868e309a6bbabfd6340c8a35